### PR TITLE
Update CHANGELOG.md and gem version for 3.0.0 release

### LINKED ISSUE #241: Update CHANGELOG.md for 3.0.0 release
As the title says.

### PR DESCRIPTION
**Pull request by Sinjo**
@dmagliola I've pulled in your work from #228, which we were going to use to cut an interim 2.2.0 release, and combined it all into one big section for the 3.0.0 release which we decided to go with instead.

Fixes #241

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,14 +1,122 @@
 # CHANGELOG
 
+# Unreleased changes
+
+_None outstanding_
```
Comment (dmagliola) [inline CHANGELOG.md:5]: Do we need this top section?

**Review by dmagliola — APPROVED** [review]
Nice :)
Looking forward to releasing this one!

File: CHANGELOG.md (hunk 1 above)
Comment (Sinjo) [reply CHANGELOG.md:5]: We don't right now. I was putting it there as a prompt for us to build this up incrementally between future releases.

Do you dislike it being there?

**Comment by Sinjo** [comment]
@dmagliola One question before we merge this: do you think it's worth us putting a release candidate out given the number of changes?

I'm thinking it might be worth it if we also post it on prometheus-announce so that there's more chance of people giving it a go. Equally, I know how reticent people are to use software that's labelled as a release candidate (but would be totally fine with using if it were a proper release), which limits their value.

**Comment by dmagliola** [comment]
I've taken a look at the changes which we introduced, and I feel like we can be fairly confident that they'll work.

I'd put out a release candidate if we ever merge lzap's change to the DirectFileStore, that kind of change seems more like something that could have hard-to-predict implications.

What we've done in these commits is the kind of thing that is pretty testable, and it's tested. I think we can release as 3.0.0, to be honest.

**Review by dmagliola — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (dmagliola) [reply CHANGELOG.md:5]: 👍

**Pull request merged**